Add documentation sections about Qiskit transpile service and AI Transpile passes (#459)
Qiskit/documentation: docs/transpile/qiskit-transpiler-service.mdx

---
title: Transpile circuits remotely with the Qiskit transpiler service
description: What is the Qiskit transpiler service and how to use it

---
# Transpile circuits remotely with the Qiskit transpiler service

> The Qiskit transpiler service provides transpilation capabilities on the cloud. In addition to the traditional [Qiskit transpiling capabilities](index), you can use IBM Quantum Cloud resources for transpilation tasks while taking advantage of AI-powered transpiler passes.

Reviewer: ```suggestion
The Qiskit transpiler service provides transpilation capabilities on the cloud. In addition to the local Qiskit transpiler capabilities, you can use the Qiskit transpiler service to use IBM Quantum Cloud resources for transpilation tasks while taking advantage of AI-powered transpiler passes.
```

Author: I think it's a good improvement (local Qiskit transpiler vs traditional)

The Qiskit transpiler service offers a Python library to seamlessly integrate this service and its capabilities into your current Qiskit patterns and workflows.

<Admonition type="note">
    This experimental service is only available for IBM Quantum Premium Plan users.
    The service is an alpha release, subject to change.
</Admonition>

<span id="install-transpiler-service"></span>
## Install the qiskit-transpiler-service package

To use the Qiskit transpiler service, install the `qiskit-transpiler-service` package:

```sh
pip install qiskit-transpiler-service
```

> By default, the package will try to authenticate to IBM Quantum services with the defined Qiskit API token, and will take your token from the `QISKIT_IBM_TOKEN` environment variable or from the file `~/.qiskit/qiskit-ibm.json` (under the section `default-ibm-quantum`).

Reviewer 2: ```suggestion
By default, the package tries to authenticate to IBM Quantum services with the defined Qiskit API token, and uses your token from the `QISKIT_IBM_TOKEN` environment variable or from the file `~/.qiskit/qiskit-ibm.json` (under the section `default-ibm-quantum`).
```

## qiskit-transpiler-service transpile options

> - `target` (optional, str) - Refers to a backend name as it would be expected by QiskitRuntimeService (i.e., `ibm_sherbrooke`). If this is set, the transpile method uses the layout from the specified backend for the transpilation operation. If any other option is set that impacts these settings, such as `coupling_map`, the `target` settings will be overridden.

Reviewer 2: ```suggestion
- `target` (optional, str) - A backend name as it would be expected by QiskitRuntimeService (for example, `ibm_sherbrooke`). If this is set, the transpile method uses the layout from the specified backend for the transpilation operation. If any other option is set that impacts these settings, such as `coupling_map`, the `target` settings are overridden.
```

> - `coupling_map` (optional, List[List[int]]) - Refers to a valid coupling map list (i.e., [[0,1],[1,2]]). If this is set, the transpile method uses this coupling map for the transpilation operation. If defined, it will override any value entered for the option `target`.

Reviewer 2: ```suggestion
- `coupling_map` (optional, List[List[int]]) - A valid coupling map list (for example, [[0,1],[1,2]]). If this is set, the transpile method uses this coupling map for the transpilation operation. If defined, it overrides any value specified for `target`.
```

> - `optimization_level` (int) - Refers to the potential optimization level to apply during the transpilation process. The valid values for this option are [1,2,3], 1 being the lightest level of optimization and 3 the top level for optimizations.

Reviewer 2: ```suggestion
- `optimization_level` (int) - The potential optimization level to apply during the transpilation process. Valid values are [1,2,3], where 1 is the least optimization (and fastest), and 3 the most optimization (and most time-intensive).
```

> - `ai` (bool) - Refers to the use of AI capabilities during the transpilation operation. The AI capabilities available can be for `AIRouting` transpiling passes or other AI synthesis methods. If the `ai` parameter is `True`, the service will apply different AI-powered transpiling passes depending on the `optimization_level` requested.

Reviewer 2: ```suggestion
- `ai` (bool) - Whether to use AI capabilities during transpilation. The AI capabilities available can be for `AIRouting` transpiling passes or other AI synthesis methods. If this value is `True`, the service applies different AI-powered transpiling passes depending on the `optimization_level` requested.
```

> - `qiskit_transpile_options` (dict) - Refers to a Python dictionary object that can include any other option that is valid in the [Qiskit `transpile()` method](defaults-and-configuration-options). If the `qiskit_transpile_options` input includes `optimization_level`, it will be discarded in favor of the `optimization_level` used as param input. If the `qiskit_transpile_options` includes any option not recognized by the Qiskit `transpile()` method, the library will raise an error.

Reviewer 2: ```suggestion
- `qiskit_transpile_options` (dict) - A Python dictionary object that can include any other option that is valid in the [Qiskit `transpile()` method](defaults-and-configuration-options). If the `qiskit_transpile_options` input includes `optimization_level`, it is discarded in favor of the `optimization_level` specified as parameter input. If the `qiskit_transpile_options` includes any option not recognized by the Qiskit `transpile()` method, the library raises an error.
```

The Qiskit transpiler service provides transpilation capabilities on the cloud.
In addition to the traditional [Qiskit transpiling capabilities](index), the Qiskit transpiler service allows users to take advantage of IBM Quantum Cloud resources to run their transpiling tasks and use the latest transpiling technology enhanced by the AI-powered transpiler passes.

The Qiskit transpiler service offers a Python library to seamlessly integrate this service and its capabilities into your current Qiskit patterns and workflows.

<Admonition type="note">
    This experimental service is only available for IBM Quantum Premium Plan users.
    The service is an alpha release, subject to change.
</Admonition>

## Install the qiskit-transpiler-service package

To use the Qiskit transpiler service, install the `qiskit-transpiler-service` package:

```sh
pip install qiskit-transpiler-service
```

By default, the package will try to authenticate to IBM Quantum services with the defined Qiskit API token, and will take your token from the `QISKIT_IBM_TOKEN` environment variable or from the file `~/.qiskit/qiskit-ibm.json` (under the section `default-ibm-quantum`).

> ### qiskit-transpiler-service transpile options

Reviewer 3: I applaud the use of the "twistie" feature :-) but since the section within the twistie isn't very long, I've removed it so that the material is more easily accessible. Feel free to reinstate the twistie if you have strong feelings about it!

Author: I'm ok! Thanks
